refinery/tool_manager/views.py

```diff
@@ -54,41 +46,82 @@ def list(self, request, *args, **kwargs):
                 ),
                 status=status.HTTP_400_BAD_REQUEST
             )
-        return super(ToolDefinitionsViewSet, self).list(request)
+        if self.request.user.has_perm('core.read_meta_dataset', self.data_set):
+            self.visualization_tools = get_objects_for_user(
+                self.request.user,
+                "tool_manager.read_visualizationtool"
+            )
+            self.workflow_tools = get_objects_for_user(
+                self.request.user,
+                "tool_manager.read_workflowtool"
+            )
+            self.user_tools.extend(self.visualization_tools)
+            self.user_tools.extend(self.workflow_tools)
+
+        else:
+            return Response(
+                "Unauthorized for data set uuid: {}".format(
+                    self.data_set.uuid
+                ),
+                status=status.HTTP_401_UNAUTHORIZED
+            )
+        return super(ToolManagerViewSetBase, self).list(request)
+
+
+class ToolDefinitionsViewSet(ToolManagerViewSetBase):
+    """API endpoint that allows for ToolDefinitions to be fetched"""
+    serializer_class = ToolDefinitionSerializer
+    lookup_field = 'uuid'
+    http_method_names = ['get']
+
+    def get_queryset(self):
+        if self.request.user.has_perm('core.share_dataset', self.data_set):
+            return ToolDefinition.objects.all()
+
+        elif self.request.user.has_perm('core.read_dataset', self.data_set):
+            return ToolDefinition.objects.filter(
+                tool_type=ToolDefinition.VISUALIZATION
+            )
```

Comment:
Is the fall-through case ok or should there be an error returned?

Reply:
@mccalluc Good point. Will discuss with @scottx611x.